File: python/sender.py
import cv2, socket, pickle

s=socket.socket(socket.AF_INET,socket.SOCK_DGRAM)
s.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 10000000)
receiverip="192.168.1.108"
receiverport=8080

cap = cv2.VideoCapture(0)
while True:
    ret,photo = cap.read()
    ret, buffer = cv2.imencode(".jpg", photo,[int(cv2.IMWRITE_JPEG_QUALITY),30])    
    x_as_bytes = pickle.dumps(buffer)    
    s.sendto(x_as_bytes,(receiverip , receiverport))    
    if cv2.waitKey(10) == 27:        
        break

cv2.destroyAllWindows()
cap.release()

File: cpp/sender.cpp
#include <arpa/inet.h>
#include <netinet/in.h>
#include <sys/socket.h>
#include <unistd.h>

#include <vector>
#include <iostream>
#include <opencv2/opencv.hpp>

int main() {
  int s;
  struct sockaddr_in serv_addr;
  cv::Mat img;
  std::vector<uchar> buf;
  cv::VideoCapture cap(0);

  s = socket(AF_INET, SOCK_DGRAM, 0);

  memset(&serv_addr, 0, sizeof(serv_addr));
  serv_addr.sin_family = AF_INET;
  serv_addr.sin_addr.s_addr = inet_addr("192.168.1.108");
  serv_addr.sin_port = htons(8080);

  while (true) {
    cap >> img;
    cv::imencode(".jpg", img, buf, {cv::IMWRITE_JPEG_QUALITY, 30});
    sendto(s, &buf[0], buf.size(), 0, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
  }

  close(s);
  cap.release();
  return 0;
}